tools/monorepo-merge/src/commands/merge/index.ts

```diff
@@ -0,0 +1,205 @@
+/**
+ * Internal dependencies
+ */
+import { MONOREPO_ROOT } from '../../const';
+import { access, exec } from '../../node-async';
+
+/**
+ * External dependencies
+ */
+import { CliUx, Command, Flags } from '@oclif/core';
+import { join } from 'path';
+import { tmpdir } from 'os';
+
+export default class Merge extends Command {
+	static description =
+		'Merges another repository into this one with history.';
+
+	static args = [
+		{
+			name: 'source',
+			description: 'The GitHub repository we are merging from.',
+			required: true,
+		},
+		{
+			name: 'destination',
+			description:
+				'The monorepo path for the repository to be merged at.',
+			required: true,
+		},
+	];
+
+	static flags = {
+		branch: Flags.string( 
+			{ 
+				description: 'The destination branch we want to merge into the monorepo.',
+				default: 'main' 
+			} 
+		)
+	};
+
+	/**
+	 * This method is called to execute the command.
+	 */
+	async run(): Promise< void > {
+		const { args, flags } = await this.parse( Merge );
+
+		await this.checkDependencies();
+		await this.validateArgs( args.source, args.destination );
+
+		let confirmation = await CliUx.ux.confirm( 'WARNING: This command will DESTROY the history of your current branch. Are you sure you want to proceed? (y/n)' );
+		if ( ! confirmation ) {
+			this.exit( 0 );
+		}
+
+		const repositoryPath = await this.cloneRepository( args.source );
+		await this.alterRepositoryHistory( args.source, repositoryPath, args.destination );
+
+		confirmation = await CliUx.ux.confirm( 'Are you ready to merge ' + args.source + ' from ' + repositoryPath + '? (y/n)' );
+		if ( ! confirmation ) {
+			// Remove the repository we've cloned.
+			try {
+				await exec( 'rm -rf ' + repositoryPath );
+			} catch {}
+			
+			this.exit( 0 );
+		}
+		
+		await this.mergeRepository( args.source, repositoryPath, flags.branch );
+
+		this.log( 'Successfully merged ' + args.source + ' into ' + args.destination );
+	}
+
+	/**
+	 * Checks to make sure that all of the necessary dependencies to run the script are installed.
+	 */
+	private async checkDependencies(): Promise< void > {
+		try {
+			await exec( 'git --version' );
+		} catch {
+			this.error( '"git" must be installed' );
+		}
+
+		try {
+			await exec( 'git-filter-repo --version' );
+		} catch {
+			this.error( '"git-filter-repo" must be installed' );
+		}
+	}
+
+	/**
+	 * Validates all of the arguments to make sure they're compatible with the command.
+	 * 
+	 * @param {string} source The GitHub repository we are merging.
+	 * @param {string} destination The local path we're merging into.
+	 */
+	private async validateArgs( source: string, destination: string ): Promise< void > {
+		// We only support pulling from GitHub so the format needs to match that.
+		if ( ! source.match( /^[a-z0-9\-]+\/[a-z0-9\-]+$/ ) ) {
```

Comment:
Probably edge-casey, and never likely to apply to us, but GitHub allows underscores, and I believe it distinguishes by case as well.